python/ray/data/dataset.py

```diff
@@ -657,6 +657,52 @@ def do_shuffle(block_list, clear_input_blocks: bool, block_udf, remote_args):
         )
         return Dataset(plan, self._epoch, self._lazy)
 
+    def randomize_block_order(
+        self,
+        *,
+        seed: Optional[int] = None,
+    ) -> "Dataset[T]":
+        """Randomly shuffle the blocks of this dataset.
+
+        Examples:
+            >>> import ray
+            >>> ds = ray.data.range(100) # doctest: +SKIP
+            >>> # Randomize the block order.
+            >>> ds.randomize_block_order() # doctest: +SKIP
+            >>> # Randomize the block order with a fixed random seed.
+            >>> ds.randomize_block_order(seed=12345) # doctest: +SKIP
+
+        Args:
+            seed: Fix the random seed to use, otherwise one will be chosen
+                based on system randomness.
+
+        Returns:
+            The shuffled dataset.
+        """
+
+        def do_shuffle(block_list, clear_input_blocks: bool, *_):
+            num_blocks = block_list.executed_num_blocks()  # Blocking.
+            if num_blocks == 0:
+                return block_list, {}
+            if clear_input_blocks:
+                blocks = block_list.copy()
+                block_list.clear()
+            else:
+                blocks = block_list
+
+            blocks.randomize_block_order(seed)
```

Comment:
I think that `ray.data.read().randomize_block_order()` will fail since `randomize_block_order()` isn't implemented for `LazyBlockList`. We could get around this by forcing execution at the beginning of this method:
```python
def randomize_block_order(...):
    ds = self.fully_executed()
    ...
```

Reply:
I think we need randomization for lazy reads too, should we also implement the random thing for LazyBlockList?

Reply:
Okay I ended up adding something naive, any suggestions on how I can test if it works?